src/outagesim/backup_reliability.jl

```diff
@@ -240,60 +466,69 @@ If ``marginal_survival`` = true then result is chance of surviving in given outa
  0.96  0.4096  0.761856
  0.64  0.8704  0.761856
  0.96  0.8704  0.761856
  0.96  0.8704  0.262144
 
-julia> survival_over_time_gen_only(critical_load, OA, failure_to_start, failure_to_run, num_generators, gen_capacity, max_duration; marginal_survival = false)
+julia> survival_over_time_gen_only(critical_load=critical_load_kw, gen_operational_availability=gen_operational_availability, 
+                                failure_to_start=failure_to_start, failure_to_run=failure_to_run, num_generators=num_generators, 
+                                gen_capacity_kw=gen_capacity_kw, max_duration=max_duration, marginal_survival = false)
 4×3 Matrix{Float64}:
  0.96  0.4096  0.393216
  0.64  0.6144  0.557056
  0.96  0.8704  0.761856
  0.96  0.8704  0.262144
 ```
 """
-                     
-function survival_over_time_gen_only(critical_load::Vector, OA::Real, failure_to_start::Real, failure_to_run::Real, num_generators::Int, 
-    gen_capacity::Real, max_duration::Int; marginal_survival = true)::Matrix{Float64} 
-
-    t_max = length(critical_load)
+function survival_over_time_gen_only(;
+    critical_load_kw::Vector, 
+    gen_operational_availability::Union{Real, Vector{<:Real}}, 
+    failure_to_start::Union{Real, Vector{<:Real}}, 
+    failure_to_run::Union{Real, Vector{<:Real}},
+    num_generators::Union{Int, Vector{Int}}, 
+    gen_capacity_kw::Union{Real, Vector{<:Real}},
+    max_duration::Int,
+    marginal_survival = true)::Matrix{Float64} 
+
+    t_max = length(critical_load_kw)
     #
-    generator_production = collect(0:num_generators).*gen_capacity
+    generator_production = generator_output(num_generators=num_generators, gen_capacity_kw=gen_capacity_kw) 
     #Initialize lost load matrix
     survival_probability_matrix = zeros(t_max, max_duration)
     #initialize amount of extra generation for each critical load hour and each amount of generators
-    generator_markov_matrix = markov_matrix(num_generators, failure_to_run)
+    generator_markov_matrix = markov_matrix(num_gen=num_generators, fail_prob=failure_to_run)
   
     #Get starting generator vector
-    starting_gens = starting_probabilities(num_generators, OA, failure_to_start) #initialize gen battery prob matrix
+    starting_gens = starting_probabilities(num_gen=num_generators, gen_operational_availability=gen_operational_availability, gen_failure_to_start=failure_to_start) #initialize gen battery prob matrix
 
-    #start loop
     for t  = 1:t_max
         gen_probs = starting_gens
         #
         for d in 1:max_duration
             survival = ones(1, length(generator_production))
             
             h = mod(t + d - 2, t_max) + 1 #determines index accounting for looping around year
-            net_gen = generator_production .- critical_load[h]
+            net_gen = generator_production .- critical_load_kw[h]
             survival[net_gen .< 0] .= 0
 
             gen_probs *= generator_markov_matrix #Update to account for generator failures
```

Comment:
Shouldn't this update not happen until the 2nd timestep because starting_probabilities() already calculated what it should be in the first tilmestep?

Reply:
@Bill-Becker could you provide your opinion on what would be correct here?

Reply:
I honestly have not navigated my way around this math to understand it; would Sean be more appropriate to answer that, or is there a debate here between you two :-)?

Reply:
the way it's being done is correct because starting_probabilities uses generator_operational_availability and generator_failure_to_start, while this loop uses generator_failure_to_run